### PR TITLE
AWK snippets remove comments

### PR DESCRIPTION
**Pull request by glennj**
Question: are existing snippets cached, or are the calculated on-the-fly?

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -24,6 +24,9 @@ Layout/MultilineMethodCallIndentation:
 Layout/EmptyLinesAroundAccessModifier:
   EnforcedStyle: only_before
 
+Layout/TrailingWhitespace:
+  AllowInHeredoc: true
```
Comment (ErikSchierboom) [inline .rubocop.yml:28]: I don't massively care, but is this necessary? Or could the trailing whitespace be removed?

**Review by ErikSchierboom — COMMENTED** [review]
> Question: are existing snippets cached, or are the calculated on-the-fly?

They're cached. We thus have to recalculate the snippets if this is merged (which is fine).

File: test/languages/awk_test.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
+require "test_helper"
+
+module SnippetExtractor
+  module Languages
+    class AwkTest < Minitest::Test
+      def test_full_example
+        code = <<~CODE
```
Comment (iHiD) [inline test/languages/awk_test.rb:7]:
```suggestion
        #rubocop:disable Layout/TrailingWhitespace
        code = <<~CODE
```

File: test/languages/awk_test.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+              RS = "^$"
+          }
+
+          END {
+              gsub(/[[:space:]]/, "")
+              if ($0 == "")
+                  print "Fine. Be that way!"
+              else {
+                  yelling = /[A-Z]/ && !/[a-z]/
+        CODE
```
Comment (iHiD) [inline test/languages/awk_test.rb:48]:
```suggestion
        CODE
        #rubocop:enable Layout/TrailingWhitespace
```

File: .rubocop.yml (hunk 1 above)
Comment (iHiD) [reply .rubocop.yml:28]: I've added two suggests to make this specific to just this HEREDOC file. I'm agnostic to the change, but I'd want to also change it in the other rubocop places (I like consistent files) and I'm not sure we necessarily want to do that now (although, I don't care if it's troublefree 🙂)

Comment (ErikSchierboom) [reply .rubocop.yml:28]: Having done that, can this be removed?

Comment (glennj) [reply .rubocop.yml:28]: d'oh!

**Review by ErikSchierboom — APPROVED** [review]

**Comment by ErikSchierboom** [comment]
@glennj I've redone the snippets for the existing iterations. Could you check to see if that worked?

**Comment by glennj** [comment]
Looks great!

**Pull request merged**